### PR TITLE
MGMT-4577 kube_helpers: Added ClusterImageSet class

### PR DESCRIPTION
**Pull request by danielerez**
Includes required changes for testing functionality implemented in https://github.com/openshift/assisted-service/pull/1319:
* Added ClusterImageSet and ClusterImageSetReference
* Created a ClusterImageSet as part of execute_day1_flow.
* Defined OPENSHIFT_VERSIONS env var in deploy_assisted_service.

### CODE REVIEW

**Comment by danielerez** [comment]
@RazRegev @YuviGold @filanov can you please take a look?

File: discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from tests.conftest import env_variables
+
+from .base_resource import BaseResource
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterImageSetReference:
+    """
+    A CRD the represents a reference for a ClusterImageSet resource.
```
Comment (YuviGold) [inline discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py:16]:
```suggestion
    A CRD that represents a reference for a ClusterImageSet resource.
```

File: discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    The reference contains only the name of the ClusterImageSet.
```
Comment (YuviGold) [inline discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py:17]: Could you elaborate what it is used for?

File: discovery-infra/test_infra/consts.py [hunk 3]
```diff
@@ -35,6 +35,7 @@
 PODMAN_FLAGS = "--cgroup-manager=cgroupfs --storage-driver=vfs --events-backend=file"
 LOG_FOLDER = "/tmp/assisted_test_infra_logs"
 DEFAULT_OPENSHIFT_VERSION = "4.7"
+DEFAULT_OPENSHIFT_INSTALL_RELEASE_IMAGE = "quay.io/openshift-release-dev/ocp-release:4.7.5-x86_64"
```
Comment (YuviGold) [inline discovery-infra/test_infra/consts.py:38]: I would prefer not to maintain another OCP version data. This default should be taken from the defaults JSON in assisted-service
 https://github.com/openshift/assisted-service/blob/4a57d53684600660d7d577e2e74b2d7e112503fe/default_ocp_versions.json#L9-L16

File: discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        data = {
+            'name': self.name,
+        }
+        return data
+
+
+class ClusterImageSet(BaseResource):
+    """
+    A CRD that represents a ClusterImageSet resource that contains the release image URI.
+    Upon creating a cluster deployment, the release image is fetched from the image set.
```
Comment (YuviGold) [inline discovery-infra/test_infra/helper_classes/kube_helpers/cluster_image_set.py:38]:
```suggestion
    Upon creating a cluster deployment, the release image is fetched by the assisted-service from the image set.
    Then, the new image will become available by the assisted-service.
```

File: scripts/deploy_assisted_service.sh [hunk 5]
```diff
@@ -15,6 +15,14 @@ export OCP_SERVICE_PORT=$(( 7000 + $NAMESPACE_INDEX ))
 export OPENSHIFT_INSTALL_RELEASE_IMAGE=${OPENSHIFT_INSTALL_RELEASE_IMAGE:-}
 export ENABLE_KUBE_API=${ENABLE_KUBE_API:-false}
 export ENABLE_KUBE_API_CMD="ENABLE_KUBE_API=${ENABLE_KUBE_API}"
+export OPENSHIFT_VERSIONS=${OPENSHIFT_VERSIONS:-"{}"}
+
+if [[ ! -z ${ENABLE_KUBE_API:-} && "${OPENSHIFT_VERSIONS}" == "{}" ]]; then
+    rhcos_image="https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/4.7.0/rhcos-4.7.0-x86_64-live.x86_64.iso"
```
Comment (YuviGold) [inline scripts/deploy_assisted_service.sh:21]: I would prefer not to maintain another OCP version data. This default should be taken from the defaults JSON in assisted-service.
https://github.com/openshift/assisted-service/blob/4a57d53684600660d7d577e2e74b2d7e112503fe/default_ocp_versions.json#L9-L16

You could use `jq` to get the entry with `default` equals to `true` and remove `release_image` fields

**Review by YuviGold — APPROVED** [review]
LGTM

**Comment by YuviGold** [comment]
It looks like there's a consistent failure with the e2e workflow. Cluster installation doesn't start (cluster is ready)
I'm holding to stop the spam.
/hold

**Review by YuviGold — APPROVED** [review]

**Pull request merged**